pootle/static/js/editor/app.js

```diff
@@ -2312,4 +2338,40 @@ PTL.editor = {
     return true;
   },
 
+  toggleSuggestion(e, canHide = false) {
+    e.stopPropagation();
+    const $suggestion = $(e.target).parents('.js-user-suggestion');
+
+    if (this.selectedSuggestionId === undefined) {
+      this.selectedSuggestionId = $suggestion.data('sugg-id');
+      const props = {
+        editor: this,
+        suggId: this.selectedSuggestionId,
+        suggText: $suggestion.data('translation-aid'),
+      };
+      const $feedbackPlaceHolder = $suggestion.find('.feedback')[0];
+      $suggestion.addClass('suggestion-expanded');
+      $('.js-editor-body .translate-full').addClass('suggestion-expanded');
+      this.suggestionFeedbackForm = ReactDOM.render(
+        <SuggestionFeedbackForm {...props} />,
+        $feedbackPlaceHolder
+      );
+      $('#suggestion-editor').focus();
+      autosize($('#suggestion-feedback-form textarea'));
+    } else if (canHide) {
+      this.closeSuggestion();
+    }
+  },
+
+  closeSuggestion(checkIfCanNavigate = true) {
+    if (this.selectedSuggestionId !== undefined &&
+        (!checkIfCanNavigate || this.canNavigate())) {
+      const $suggestion = $(`#suggestion-${this.selectedSuggestionId}`);
+      this.selectedSuggestionId = undefined;
+      this.suggestionFeedbackForm = undefined;
```

Comment:
In order to unmount a component, the [`ReactDOM.unmountComponentAtNode()`](https://facebook.github.io/react/docs/top-level-api.html#reactdom.unmountcomponentatnode) API must be used, or it will potentially leak memory.
At the same time, if the suggestion was left open and the user navigates away from the unit, the component will leak memory too. This is already handled in the [special font PR](https://github.com/translate/pootle/pull/4171/commits/fe95ef0169c752d19090fa16ae64039fda1d71ab).